scripts/addr2line.py

```diff
@@ -83,41 +84,148 @@ def __call__(self, address):
+            # we use a bit of a quick and dirty heuristic: if the symbol is *far enough* beyond
+            # the last symbol we assume it is not valid. Most likely, the overwhemling majority
+            # of cases are invalid (e.g., due to KASLR) as the final symbol in the map is usually
+            # something obscure.
+            lastsym = sa[-1]
+            if address - lastsym > self.LAST_SYMBOL_MAX_SIZE:
+                return f'{addrstr} ({address - lastsym} bytes after last symbol)\n'
+        saddr = sa[idx]
+        assert saddr <= address
+        return '{}+0x{:x}\n'.format(sn[idx], address - saddr)
+
+
 class BacktraceResolver(object):
 
     class BacktraceParser(object):
         class Type(Enum):
             ADDRESS = 1
             SEPARATOR = 2
+            KERNEL_ADDRESS = 3
 
         def __init__(self):
             addr = "0x[0-9a-f]+"
-            path = "\S+"
-            token = f"(?:{path}\+)?{addr}"
-            full_addr_match = f"(?:({path})\+)?({addr})"
-            self.oneline_re = re.compile(f"^((?:.*(?:(?:at|backtrace):?|:))?(?:\s+))?({token}(?:\s+{token})*)(?:\).*|\s*)$", flags=re.IGNORECASE)
+            path = r"\S+"
+            token = fr"(?:{path}\+)?{addr}"
+            full_addr_match = fr"(?:({path})\+)?({addr})"
+            self.oneline_re = re.compile(
+                fr"^((?:.*(?:(?:at|backtrace):?|:))?(?:\s+))?({token}(?:\s+{token})*)(?:\).*|\s*)$",
+                flags=re.IGNORECASE)
             self.address_re = re.compile(full_addr_match, flags=re.IGNORECASE)
-            self.asan_re = re.compile(f"^(?:.*\s+)\(({full_addr_match})\)\s*$", flags=re.IGNORECASE)
-            self.separator_re = re.compile('^\W*-+\W*$')
+            self.asan_re = re.compile(
+                fr"^(?:.*\s+)\(({full_addr_match})\)\s*$", flags=re.IGNORECASE)
+            self.kernel_re = re.compile(fr'^kernel callstack: (?P<addrs>(?:{addr}\s*)+)$')
+            self.separator_re = re.compile(r'^\W*-+\W*$')
+
+
+        def split_addresses(self, addrstring: str, default_path=None):
+            addresses : list[dict[str, Any]] = []
+            for obj in addrstring.split():
+                m = re.match(self.address_re, obj)
+                assert m, f'addr did not match address regex: {obj}'
+                #print(f"  >>> '{obj}': address {m.groups()}")
+                addresses.append({'path': m.group(1) or default_path, 'addr': m.group(2)})
+            return addresses
 
         def __call__(self, line):
             def get_prefix(s):
                 if s is not None:
                     s = s.strip()
                 return s or None
 
+            # order here is important: the kernel callstack regex
+            # needs to come first since it is more specific and would
+            # otherwise be matched by the online regex which comes next
+            m = self.kernel_re.match(line)
+            if m:
+                # print(f">>> '{line}': {m.groups()}")
```

Comment:
Dead commented code

Reply:
Here I followed the pattern of the existing matchers which all included this commented out debugging code, but I will delete it.